presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -742,6 +749,81 @@ public void testValidateFileBuckets()
         BackgroundHiveSplitLoader.validateFileBuckets(bucketFiles, 30, "tableName", "partitionName");
     }
 
+    @Test
+    public void testBuildManifestFileIterator()
+            throws Exception
+    {
+        CachingDirectoryLister directoryLister = new CachingDirectoryLister(new Duration(5, TimeUnit.MINUTES), 1000, ImmutableList.of());
+        Configuration configuration = new Configuration(false);
+        Properties schema = new Properties();
+        schema.setProperty(FILE_INPUT_FORMAT, SymlinkTextInputFormat.class.getName());
+        schema.setProperty(SERIALIZATION_LIB, AVRO.getSerDe());
+
+        Path firstFilePath = new Path("hdfs://VOL1:9000/db_name/table_name/file1");
+        Path secondFilePath = new Path("hdfs://VOL1:9000/db_name/table_name/file2");
+        List<Path> paths = ImmutableList.of(firstFilePath, secondFilePath);
+        List<LocatedFileStatus> files = paths.stream()
+                .map(TestBackgroundHiveSplitLoader::locatedFileStatus)
+                .collect(toImmutableList());
+
+        BackgroundHiveSplitLoader backgroundHiveSplitLoader = backgroundHiveSplitLoader(
+                files,
+                directoryLister);
+        Optional<Iterator<InternalHiveSplit>> splitIterator = backgroundHiveSplitLoader.buildManifestFileIterator(
+                configuration,
+                "partition",
+                schema,
+                ImmutableList.of(),
+                TupleDomain.all(),
+                () -> true,
+                false,
+                TableToPartitionMapping.empty(),
+                new Path("hdfs://VOL1:9000/db_name/table_name"),
+                paths,
+                true);
+        assertTrue(splitIterator.isPresent());
+        List<InternalHiveSplit> splits = Streams.stream(splitIterator.get())
+                .collect(toImmutableList());
+        assertEquals(splits.size(), 2);
+        assertEquals(splits.get(0).getPath(), firstFilePath.toString());
```

Comment:
The test relies on ordering in `paths` being preserved. Maybe use `assertEqualsNoOrder(paths, Streams.stream(splitIterator.get()).map(InternalHiveSplit::getPath).collect(toImmutableList())`?